### PR TITLE
Error early if rows could not resolve to anything in data_color()

### PR DESCRIPTION
**Pull request by olivroy**
# Summary

Consider this,

```r
mtcars |>
  gt() |>
  data_color(rows = cyl == 5)
```

Currently, this errors with a cryptic error:

```r
Error in `get_contrast_ratio()`:
! At least one color defined for each of
  `color_1` and `color_2`.
```

I have left a review with an alternative approach! Please let me know which one you think is better

# Related GitHub Issues and PRs

Discovered while answering #1659 

# Checklist

- [x] I understand and agree to the [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html).
- [x] I have listed any major changes in the [NEWS](https://github.com/rstudio/gt/blob/master/NEWS.md).
- [x] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/rstudio/gt/tree/master/tests/testthat) for any new functionality.

### CODE REVIEW

**Review by olivroy — COMMENTED** [review]
Alternative PR! (would not be a breaking change, since this currently errors)

Maybe less suprising this way

**Comment by rich-iannone** [comment]
This is a bit tricky API-wise. I like the erroring if you supply a specific row index, row name, or set of them. However, if providing an expression it would be nice to no-op, which is just like the behavior for columns. For rows, I don't know that we currently make the distinction between expression and explicit selection. Also, I don't know if this is the current behavior for other functions with a `rows` argument (mostly `fmt_*()` and IIRC some `cols_*()` fns too).

So, it's worth discussing this a little! I know what I proposed would be a breaking change but I hope it would be a welcome one that fixes some inconsistencies.

**Comment by olivroy** [comment]
Good idea!

In this case, let's go with the alternative PR (that is the status quo)? and I will open a new issue to do that in a folow-up PR.

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # gt (development version)
 
+* `data_color()` throws a more informative error message if `rows didn't resolve to anything (@olivroy, #1659).
```
Comment (rich-iannone) [inline NEWS.md:3]:
```suggestion
* `data_color()` throws a more informative error message if `rows` didn't resolve to anything (@olivroy, #1659).
```

**Comment by rich-iannone** [comment]
Disregard the above suggestion, as we're going with a different PR (feel free to close this one).

**Comment by olivroy** [comment]
Will you open a new PR?  The "alternative" PR is this PR in its current state, after I accepted the suggestions https://github.com/rstudio/gt/pull/1660#pullrequestreview-2053232141.

My suggestion would be to merge this which basically just changes

```r
# before
mtcars |>
  gt() |>
  data_color(rows = cyl == 5)

Error in `get_contrast_ratio()`:
! At least one color defined for each of
  `color_1` and `color_2`.

# this PR
mtcars |>
  gt() |>
  data_color(rows = cyl == 5)

Error in `data_color()`:
!  `rows` resulted in an empty selection.
```

And address outstanding issue in a follow-up (issue #1665 )?

Does that make sense?

**Comment by rich-iannone** [comment]
Yes, it makes sense! (Sorry for the confusion here.)

**Review by rich-iannone — APPROVED** [review]
LGTM!

**Pull request merged**